### PR TITLE
[GPU] Binary post-op support for full tensor.

### PR DESCRIPTION
**Pull request by jade-cho**
Ticket
- 63397

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #include "intel_gpu/runtime/debug_configuration.hpp"
 #include "program_helpers.h"
 #include "binary_convolution_inst.h"
+#include "resample_inst.h"
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp:14]: I will remove this line.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 2]
```diff
@@ -612,6 +613,55 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
                 conv_node.get_dependencies().front()->set_output_layout(new_layout, false);
             }
         }
+
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp:616]: got it.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 3]
```diff
@@ -612,6 +613,55 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
...
+        if (lo.get_optimization_attributes().use_onednn_impls) {
+            std::vector<program_node*> prev_nodes;
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp]: I think it's good.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 4]
```diff
@@ -612,6 +613,55 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
...
+
+            // key: conv format, conv data type, dependency data type, value: dependency format
+            std::map<std::tuple<format, data_types, data_types>, format> expected_df_map = {
+                {{format::b_fs_yx_fsv16, data_types::f16, data_types::u8},  format::b_fs_yx_fsv32},
+                {{format::b_fs_yx_fsv16, data_types::f16, data_types::f16}, format::b_fs_yx_fsv16},
+                {{format::b_fs_yx_fsv16, data_types::f16, data_types::f32}, format::b_fs_yx_fsv16},
+                {{format::bs_fs_yx_bsv32_fsv16, data_types::f16, data_types::u8},  format::bs_fs_yx_bsv32_fsv32},
+                {{format::bs_fs_yx_bsv32_fsv16, data_types::f16, data_types::f16}, format::bs_fs_yx_bsv32_fsv16},
+                {{format::bs_fs_yx_bsv32_fsv16, data_types::f16, data_types::f32}, format::bs_fs_yx_bsv32_fsv16}
+            };
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp]: changed

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 5]
```diff
@@ -612,6 +613,55 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
...
+                        auto new_layout = old_layout;
+                        new_layout.format = expected_format;
+                        auto new_input = rf.get_reorder(prev_node->id(), old_layout, new_layout);
+                        // if (new_input.first) {
+                        //     std::cout << __FILE__ << ":" << __LINE__ << " " << __func__ << " "
+                        //         << conv_node.id() << " " << prev_node->id() << std::endl;
+                        //     p.add_intermediate(new_input.first, conv_node, 0, !new_input.second);
+                        // }
+                        prev_node->set_output_layout(new_layout, false);
+                    }
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp:651]: changed

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 6]
```diff
@@ -806,11 +807,11 @@ void program_node::init_onednn_primitive_attributes() {
             auto in = get_dependency(dep_idx).get_output_layout();
 
             if (e_node.get_primitive()->mode == eltwise_mode::sum) {
-                if (e_node.get_primitive()->needs_onednn_sum_post_op(in)) {
+                if (program_helpers::needs_onednn_sum_post_op(e_node, in)) {
                     post_ops.append_sum(1.0f, onednn::convert_data_type(in.data_type));
                     update_onednn_post_op_list(onednn_post_op_type::sum, dep_idx);
                 } else {
-                    dnnl::memory::desc in_desc = onednn::layout_to_memory_desc(in, dnnl::memory::format_tag::ab, true);
+                    dnnl::memory::desc in_desc = onednn::layout_to_memory_desc(in);
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:814]: won't change this.

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 7]
```diff
@@ -806,11 +807,11 @@ void program_node::init_onednn_primitive_attributes() {
             auto in = get_dependency(dep_idx).get_output_layout();
 
             if (e_node.get_primitive()->mode == eltwise_mode::sum) {
-                if (e_node.get_primitive()->needs_onednn_sum_post_op(in)) {
+                if (program_helpers::needs_onednn_sum_post_op(e_node, in)) {
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:810]: I will create the other PR for unit tests.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp [hunk 8]
```diff
@@ -612,6 +613,55 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
                 conv_node.get_dependencies().front()->set_output_layout(new_layout, false);
             }
         }
+
+        if (lo.get_optimization_attributes().use_onednn_impls) {
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp:617]: I will change.

**Review by isanghao — COMMENTED** [review]
LGTM. Let's have some more perf review.

**Review by isanghao — APPROVED** [review]

**Pull request merged**